state.go

```diff
@@ -21,6 +21,61 @@ import (
 	"github.com/pkg/errors"
 )
 
+// State transitions:
+//
+//                             init
+//
+//                               |        (wait timeout expired)
+//                               |   +---------------------------------+
+//                               |   |                                 |
+//                               v   v                                 |
+//                                         (auth req. failed)
+//                          bootstrapped ----------------------> authorize wait
+//
+//                                |
+//                                |
+//                                |  (auth data avail.)
+//                                |
+//                                v
+//
+//                           authorized
+//
+//          (update needs     |   |
+//           commit)          |   |
+//         +------------------+   |
+//         |                      |
+//         v                      |          (wait timeout expired)
+//                                |    +-----------------------------+
+//   update commit                |    |                             |
+//                                v    v                             |
+//         |                                (no update)
+//         +---------------> update check ---------------->  update check wait
+//
+//                                |
+//                                | (update ready)
+//                                v
+//
+//                           update fetch
+//
+//                                |
+//                                | (update fetched)
+//                                v
+//
+//                          update install
+//
+//                                |
+//                                | (update installed,
+//                                |  enabled)
+//                                |
+//                                v
+//
+//                              reboot
+//
+//                                |
+//                                v
+//
+//                              final (daemon exit)
```

Comment:
Do we really exit after the reboot stage? IMHO init scripts should take care of this and kill us afterwards. This might be useful later if we add logic to detect failed reboots (forcing unclean reboot if a process is hanging, for example). Just a thought.